src/osd/SnapMapper.cc

```diff
@@ -972,4 +992,65 @@ int SnapMapper::convert_legacy(
   }
   return 0;
 }
+
+
+int SnapMapper::convert_malformed(
+  CephContext *cct,
+  ObjectStore *store,
+  ObjectStore::CollectionHandle& ch,
+  ghobject_t hoid,
+  unsigned max)
+{
+  dout(10) << __func__ << "in" << dendl;
+  uint64_t n = 0;
+  std::set<std::string> old_keys;
+
+  ObjectMap::ObjectMapIterator iter = store->get_omap_iterator(ch, hoid);
+  if (!iter) {
+    return -EIO;
+  }
+
+  auto start = ceph::mono_clock::now();
+
+  iter->upper_bound(SnapMapper::MAPPING_PREFIX);
+  map<string,ceph::buffer::list> to_set;
+  while (iter->valid()) {
+    bool valid = SnapMapper::is_malformed_mapping(iter->key());
```

Comment:
A confusing collision of naming:
'is_malformed..()' returns true for 'valid' (not malformed)?